wave_vision/src/dataset.cpp

```diff
@@ -0,0 +1,253 @@
+#include "wave/vision/dataset.hpp"
+
+namespace wave {
+
+bool VOTestCamera::update(double dt) {
+    this->dt += dt;
+
+    if (this->dt > (1.0 / this->hz)) {
+        this->dt = 0.0;
+        this->frame++;
+        return true;
+    }
+
+    return false;
+}
+
+int VOTestCamera::checkFeatures(double dt,
+                                const MatX &features,
+                                const Vec3 &rpy,
+                                const Vec3 &t,
+                                std::vector<std::pair<Vec2, Vec3>> &observed) {
+    // pre-check
+    if (this->update(dt) == false) {
+        return -1;  // not time to check yet
+    }
+
+    // create rotation matrix from roll pitch yaw
+    Mat3 R;
+    euler2rot(rpy, 123, R);
+
+    // projection matrix
+    MatX P;
+    projection_matrix(this->K, R, -R * t, P);
+
+    // check which features in 3d are observable from camera
+    observed.clear();
+    for (int i = 0; i < features.cols(); i++) {
+        // convert feature in NWU to EDN coordinate system
+        Vec4 f_3d, f_3d_edn;
+        f_3d = features.block(0, i, 4, 1);
+        f_3d_edn(0) = -f_3d(1);
```

Comment:
@chutsu I see you've added the NWU to EDN function above...why are you doing this manually here again?